src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -60,6 +60,42 @@
         # Ex: select `delimited `` with escaped` from `just delimited`
         # https://spark.apache.org/docs/latest/sql-ref-identifier.html#delimited-identifier
         RegexLexer("back_quote", r"`([^`]|``)*`", CodeSegment),
+        # Numeric literal matches integers, decimals, and exponential formats.
+        # https://spark.apache.org/docs/latest/sql-ref-literals.html#numeric-literal
+        # Pattern breakdown:
```

Comment:
Nice work! I'm curious, could we potentially use Python "verbose regular expressions"? I have never used them, but IIUC, it would let us write the actual code in something similar to the format of your comments here.

https://www.geeksforgeeks.org/verbose-in-python-regex/

Reply:
yeah good shout, I've never used the flag before but will experiment 👍